docs/InfiniteList.md

```diff
@@ -0,0 +1,123 @@
+| `hasCreate`                | Optional | `boolean`      | `false`                 | Set to `true` to show the create button.                                                     |
+| `pagination`               | Optional | `ReactElement` | `<Infinite Pagination>` | The pagination component to use.                                                             |
+| `perPage`                  | Optional | `number`       | `10`                    | The number of records to fetch per page.                                                     |
+| `queryOptions`             | Optional | `object`       | -                       | The options to pass to the `useQuery` hook.                                                  |
+| `resource`                 | Optional | `string`       | -                       | The resource name, e.g. `posts`.                                                             |
+| `sort`                     | Optional | `object`       | -                       | The initial sort parameters.                                                                 |
+| `storeKey`                 | Optional | `string`       | -                       | The key to use to store the current filter & sort.                                           |
+| `title`                    | Optional | `string`       | -                       | The title to display in the App Bar.                                                         |
+| `sx`                       | Optional | `object`       | -                       | The CSS styles to apply to the component.                                                    |
+
+Check the [`<List>` component](./List.md) for details about each prop.
+
+Additional props are passed down to the root component (a MUI `<Card>` by default).
+
+## Showing The Record Count
+
+One drawback of the `<InfiniteList>` component is that it doesn't show the number of results. To fix this, you can use `useListContext` to access the `total` property of the list, and render the total number of results in a sticky footer:
+
+![Infinite list with total number of results](./img/infinite-pagination-count.webp)
+
+{% raw %}
+```jsx
+import { useListContext, InfinitePagination, InfiniteList } from 'react-admin';
+import { Box, Card, Typography } from '@mui/material';
+
+const CustomPagination = () => {
+    const { total } = useListContext();
+    return (
+        <>
+            <InfinitePagination />
+            {total > 0 && (
+                <Box position="sticky" bottom={0} textAlign="center">
+                    <Card
+                        elevation={2}
+                        sx={{ px: 2, py: 1, mb: 1, display: 'inline-block' }}
+                    >
+                        <Typography variant="body2">{total} results</Typography>
+                    </Card>
+                </Box>
+            )}
+        </>
+    );
+};
```

Comment:
Because I don't think this is a general solution, it's just an example. People will want to put the number of results on top of the list, in a datagrid footer, in the action bar, etc... There is no standard for that UI.